plugin/completion.py

```diff
@@ -255,10 +255,11 @@ def cancel_async(self) -> None:
         self._cancel_pending_requests_async()
 
     def _cancel_pending_requests_async(self) -> None:
-        for request_id, weak_session in self._pending_completion_requests.items():
-            if session := weak_session():
-                session.cancel_request(request_id, False)
+        pending_requests = self._pending_completion_requests.copy()
         self._pending_completion_requests.clear()
```

Comment:
I think this line is also not necessary anymore, because the pending requests should all be popped when the response handler is called (with an error response from cancel_request_async). Then you could even save another line by putting `self._pending_completion_requests.copy().items()` directly in the for-loop.